File: src/common/web/data/entities/connector/categories/ArchiveConnectorCategory.ts
import { ConnectorCategory, type ConnectorCategoryID } from "./ConnectorCategory";

export class ArchiveConnectorCategory extends ConnectorCategory {
    public static readonly CategoryID: ConnectorCategoryID = "archive";

    public constructor() {
        super("Archive", "Store your project data in a long-term external storage archive.");
    }
}

File: src/common/py/data/entities/connector/categories/archive_connector_category.py
import typing
from dataclasses import dataclass

from .connector_category import ConnectorCategory, ConnectorCategoryID
from .connector_categories_catalog import ConnectorCategoriesCatalog


@dataclass(frozen=True, kw_only=True)
class ArchiveConnectorCategory(ConnectorCategory):
    category_id: typing.ClassVar[ConnectorCategoryID] = "archive"

    def __init__(self):
        super().__init__(
            name="Archive",
            description="Store your project data in a long-term external storage archive.",
        )


ConnectorCategoriesCatalog.register_item(
    ArchiveConnectorCategory.category_id, ArchiveConnectorCategory()
)
